keras/src/models/model.py

```diff
@@ -204,6 +213,11 @@ def get_layer(self, name=None, index=None):
                 return self.layers[index]
 
         if name is not None:
+            # Check if the name matches any of the input names.
```

Comment:
I don't think this is needed, the fallback should work for this use case.

Also, coverage shows it's never used:
https://app.codecov.io/gh/keras-team/keras/pull/20842?src=pr&el=tree&filepath=keras%2Fsrc%2Fmodels%2Fmodel.py&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=keras-team#b343e88c9bde68f6f2b9d00f13fa225e-R218